scrollingpagerindicator/src/main/java/ru/tinkoff/scrollingpagerindicator/RecyclerViewAttacher.java

```diff
@@ -136,9 +163,12 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
 
     @Override
     public void detachFromPager() {
-        attachedAdapter.unregisterAdapterDataObserver(dataObserver);
+        adapterDelegate.unregisterAdapterDataObserver(dataObserver);
         recyclerView.removeOnScrollListener(scrollListener);
         measuredChildWidth = 0;
+        if (adapterDelegate instanceof DefaultRecyclerViewAdapterDelegate) {
```

Comment:
This condition looks useless.

Reply:
If we remove this check then in case if RecyclerViewAttacher was created with custom implementation of RecyclerViewAdapterDelegate reattach to pager recreates it with DefaultRecyclerViewAdapterDelegate.

So, I will add setAdapterDelegate(RecyclerViewAdapterDelegate adapterDelegate) for such cases.